### ISSUE TITLE
Problem: workflow content types are still present in the database

### ISSUE THREAD

**Issue by scollazo** [archivematica/Issues#514]
**Expected behaviour**
When running the database update, all interactive questions can be avoided using --no-input , and some tables are left behind

**Current behaviour**
```
archivematica@amupdate-bionic:/usr/share/archivematica/dashboard$ /usr/share/archivematica/virtualenvs/archivematica-dashboard/bin/python manage.py migrate --noinput
Operations to perform:
  Synchronize unmigrated apps: django_forms_bootstrap, staticfiles, messages, webdesign, accounts, installer, mcp
  Apply all migrations: sessions, administration, tastypie, auth, contenttypes, fpr, longerusername, main
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
  Installing custom SQL...
Running migrations:
  No migrations to apply.
```
But if you run it again, without --no-migrations, you get:

```
archivematica@amupdate-bionic:/usr/share/archivematica/dashboard$ /usr/share/archivematica/virtualenvs/archivematica-dashboard/bin/python manage.py migrate 
Operations to perform:
  Synchronize unmigrated apps: django_forms_bootstrap, staticfiles, messages, webdesign, accounts, installer, mcp
  Apply all migrations: sessions, administration, tastypie, auth, contenttypes, fpr, longerusername, main
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
  Installing custom SQL...
Running migrations:
  No migrations to apply.
The following content types are stale and need to be deleted:

    main | taskconfigunitvariablelinkpull
    main | taskconfig
    main | watcheddirectoryexpectedtype
    main | microservicechainlinkexitcode
    main | microservicechainchoice
    main | standardtaskconfig
    main | taskconfigsetunitvariable
    main | microservicechainlink
    main | watcheddirectory
    main | microservicechoicereplacementdic
    main | tasktype
    main | microservicechain

Any objects related to these content types by a foreign key will also
be deleted. Are you sure you want to delete these content types?
If you're unsure, answer 'no'.

    Type 'yes' to continue, or 'no' to cancel:
```

And this prompt can't be avoided.

**Steps to reproduce**

On a AM 1.8 install upgraded to AM 1.9, run the above commands.

**Your environment (version of Archivematica, OS version, etc)**

AM 1.9 in bionic, updated from AM 1.8.1.

---
**For Artefactual use:**
Please make sure these steps are taken before moving this issue from Review to Verified in Waffle:

- All PRs related to this issue are properly linked 👍
- All PRs related to this issue have been merged 👍
- Test plan for this issue has been implemented and passed 👍
- Documentation regarding this issue has been written and it has been added to the release notes, if needed 👍

**Comment by sevein** [comment]
> And this prompt can't be avoided.

I think that this is by design. Django 1.8 does not remove unused `contenttypes` when `--noinput` is used, but it prefers to ask you when it's not. To be clear, `contenttypes` are not tables. I think the latter should have been removed by now.

> Problem: database upgrade can't be run interactively

Perhaps this is not entirely accurate. You can still run it interactively but not until you decide what to do with the `contenttypes` lying around.

There's `remove_stale_contenttypes` in Django 1.11 but we're not there yet.

**Comment by cole** [comment]
It should be possible (but maybe not desirable or worthwhile - depends on the seriousness of this issue) to write a data migration that would remove the content types rows as part of the first command run in the issue description (assuming these are models that have been deleted).

**Comment by sevein** [comment]
I'd be happy to update [0067](https://github.com/artefactual/archivematica/blob/qa/1.x/src/dashboard/src/main/migrations/0067_delete_workflow_models.py) too, would that be a good idea @cole? We've always accepted changes in existing migrations as long as they don't belong a final release, e.g. 0067 happened after 1.8.1 was released but 1.9.0 is still in RC.

**Comment by sevein** [comment]
FYI 0067 is the migration that removed the workflow models we were discussing yesterday.

**Comment by cole** [comment]
Oh yeah, cool 😎!

I would recommend doing it as a new migration (0068, I guess) in order to keep the data migration (deleting the content type rows) separate from the schema migration (dropping the old tables). I don't think Django enforces this, they do recommend it though:

>Migrations that alter data are usually called “data migrations”; they’re best written as separate migrations, sitting alongside your schema migrations.

Source: https://docs.djangoproject.com/en/2.2/topics/migrations/#data-migrations

If that is a good idea at this point in the release process — to me it depends on how much of a pain saying 'yes' at the command prompt is. If that breaks things for a lot of people/makes administration harder, then maybe it's worthwhile.

I do think the extra migration is pretty low risk as long as you write it so that it handles both cases where the content types exist, and that they don't (i.e. handle DoesNotExist or just use the manager delete method).

**Comment by sevein** [comment]
> Migrations that alter data are usually called “data migrations”; they’re best written as separate migrations, sitting alongside your schema migrations.

I totally missed that, we've been doing it wrong for a long time.

I will do that, thanks.

**Comment by replaceafill** [comment]
This issue is being closed as part of a bulk closure process. Please click [here](https://groups.google.com/u/1/g/archivematica/c/Sq7uw1zloTA) for further information.
